### DISCUSSION TITLE
Request for Comment: Towards mst-gql v1.0.0

### DISCUSSION THREAD

**Discussion by jesse-savary** [mobxjs/mst-gql#418] [Announcements]
Hey everyone, we're super excited to be working on mst-gql v1.0.0, and we'd love to get your input on what should be included in this major release. We want to make sure we're addressing the most important features and improvements that will make a real difference for you, our users.

So, we'd like you to share your thoughts on the following:

- What new features would you love to see in mst-gql v1.0.0?
- Are there any existing issues or pain points that you'd like us to prioritize and address in this release?
- What improvements or enhancements would make your experience using mst-gql even better?

Don't hold back – we're eager to hear your ideas, suggestions, and any feedback you may have. Your input will play a crucial role in shaping the future of mst-gql and making v1.0.0 a success for everyone.

**Comment by beepsoft** [comment]
There are some ideas in older issues, which worth reconsidering:

* A new generator. https://github.com/mobxjs/mst-gql/issues/13  The current generator, just like the whole project was more of a PoC by @mweststrate  therefore the generator is not very polished and we already added bunch of stuff to it here and there, which makes it hard to maintain. It would need a rewrite, possibly in typescript. There's a new project called [mst-query](https://github.com/ConrabOpto/mst-query), which has already [done that](https://github.com/ConrabOpto/mst-query/tree/main/packages/mst-query-generator).  Its generator is based on mst-gql and practically generates the same model files. I raised an issue https://github.com/ConrabOpto/mst-query/issues/14 whether it would be possible to share the generator between the two projects but the author doesn't support it. Nevertheless, we can copy and reuse it in mst-gql as our base (both projects use MIT License).

* Batched queries. https://github.com/mobxjs/mst-gql/issues/4 Apollo client has this feature as well. It would be great if mst-gql could be positioned overall as an alternative to Apollo client.

* Suspense support. https://github.com/mobxjs/mst-gql/issues/134

Some new ideas:

* Integrations with other libraries. This should not be a core feature of mst-gql but would be great to provide examples how it can be integrated with other (query) libraries. For example, I've been experimenting integrating mst-gql with https://gqty.dev/. With gqty one can have a nicer query interface (way faster autocomplete than with mst-gql) and automatic query batching while still using mst-gql's query/caching/storage mechanisms. 

* Use a CI tool like CircleCI to better track the health of the project

* Tests with more complex (eg. Hasura generated) graphql schemas. This would show the weak points when working with large models.

Pain points:

* In case of huge graphql models, like those generated by Hasura the autocompletion for the mst-gql generated client in IDEs is very slow, sometimes nearly unusable. Not sure whether we can do anything about it, but it is definitely a pain point.

**Comment by beepsoft** [comment]
A nice review and a list of pain points:

https://medium.com/better-programming/two-years-with-mst-gql-in-production-c278fef6123e
